kafka-ballerina/tests/consumer_client_tests.bal

```diff
@@ -0,0 +1,1088 @@
+    ConsumerRecord[] consumerRecords = check consumer->poll(5);
+    test:assertEquals(consumerRecords.length(), 5, "Expected: 5. Received: " + consumerRecords.length().toString());
+    check sendMessage(TEST_MESSAGE.toBytes(), topic4);
+    check consumer->seekToEnd([topicPartition]);
+    consumerRecords = check consumer->poll(5);
+    test:assertEquals(consumerRecords.length(), 0, "Expected: 0. Received: " + consumerRecords.length().toString());
+
+    (Error|error)? result = trap consumer->seekToEnd([nonExistingPartition]);
+    if (result is Error) {
+        string expectedErrorMsg = "Failed to seek the consumer to the end: No current " +
+            "assignment for partition non-existing-topic-999";
+        test:assertEquals(result.message(), expectedErrorMsg);
+    } else {
+        test:assertFail(msg = "Expected an error");
+    }
+
+    check consumer->close();
+}
+
+@test:Config {
+    dependsOn: [consumerSeekTest, consumerSeekToBeginningTest, consumerSeekToEndTest]
+}
+function consumerPositionOffsetsTest() returns error? {
+    ConsumerConfiguration consumerConfiguration = {
+        offsetReset: OFFSET_RESET_EARLIEST,
+        groupId: "consumer-position-offset-test-group",
+        clientId: "test-consumer-18"
+    };
+    TopicPartition topicPartition = {
+        topic: topic4,
+        partition: 0
+    };
+    Consumer consumer = check new (DEFAULT_URL, consumerConfiguration);
+    check consumer->assign([topicPartition]);
+    int|Error partitionOffsetBefore = consumer->getPositionOffset(topicPartition);
+    if (partitionOffsetBefore is error) {
+        test:assertFail(msg = "Invalid offset received");
+    } else {
+        test:assertEquals(partitionOffsetBefore, 0, "Expected: 0. Received: " + partitionOffsetBefore.toString());
+    }
+    ConsumerRecord[] consumerRecords = check consumer->poll(5);
+    int|Error partitionOffsetAfter = consumer->getPositionOffset(topicPartition, TIMEOUT_DURATION);
+    if (partitionOffsetAfter is error) {
+        test:assertFail(msg = "Invalid offset received");
+    } else {
+        test:assertEquals(partitionOffsetAfter, 6, "Expected: 6. Received: " + partitionOffsetAfter.toString());
+    }
+    check consumer->close();
+    consumerConfiguration = {
+        offsetReset: OFFSET_RESET_EARLIEST,
+        groupId: "consumer-position-offset-test-group",
+        clientId: "test-consumer-19",
+        defaultApiTimeout: DEFAULT_TIMEOUT
+    };
+    consumer = check new (DEFAULT_URL, consumerConfiguration);
+    check consumer->assign([topicPartition]);
+    check sendMessage(TEST_MESSAGE.toBytes(), topic4);
+    consumerRecords = check consumer->poll(5);
+    partitionOffsetAfter = consumer->getPositionOffset(topicPartition);
+    if (partitionOffsetAfter is error) {
+        test:assertFail(msg = "Invalid offset received");
+    } else {
+        test:assertEquals(partitionOffsetAfter, 7, "Expected: 7. Received: " + partitionOffsetAfter.toString());
+    }
```

Comment:
Check for similar cases.

Basically, if the function call can return an error, and you do not need it to be an error, use the `check` expression to return the error directly. No need of type-guarding here.

```suggestion
    partitionOffsetAfter = check consumer->getPositionOffset(topicPartition);
    test:assertEquals(partitionOffsetAfter, 7, "Expected: 7. Received: " + partitionOffsetAfter.toString());
```